### PR TITLE
add checklist feature

### PR DESCRIPTION
**Pull request by JosueRodriguezGarcia**

### CODE REVIEW

File: src/main/java/trello/ui/pages/BoardPage.java [hunk 1]
```diff
@@ -61,6 +62,12 @@
     @FindBy(className = "js-add-a-card")
     private WebElement addACard;
 
+    @FindBy(className = "js-card-details")
+    private List<WebElement> cards;
+
+    @FindBy(className = "js-open-add-list")
+    private WebElement addAnotherLits;
```
Comment (MelissaRoman06) [inline src/main/java/trello/ui/pages/BoardPage.java]: Please fix the typo in the variable name

File: src/main/java/trello/ui/pages/modal/CardModal.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.List;
+
+/**
+ * CardModal class.
+ *
+ * @author Josue Rodriguez Garcia.
+ * @version 0.0.1
+ */
+public class CardModal extends BasePage {
+    @FindBy(className = "js-card-detail-title-input")
```
Comment (raulchoque-eng) [inline src/main/java/trello/ui/pages/modal/CardModal.java:31]: Please, add a white line above this line.

File: src/main/java/trello/ui/pages/modal/CardModal.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        boolean result = false;
+        for (WebElement checkList : checkLists) {
+            if (checkList.findElement(By.cssSelector("h3.hide-on-edit")).getText().equals(checkListTitle)) {
+                result = true;
+            }
+        }
+        return result;
+    }
+
+
```
Comment (raulchoque-eng) [inline src/main/java/trello/ui/pages/modal/CardModal.java]: Please, clear this white line.

File: src/main/java/trello/ui/pages/modal/CheckListModal.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    /**
+     * Does click the "add button" for confirm.
+     */
+    public void clickConfirmAddCheckListButton() {
+        confirmAddCheckListButton.click();
+    }
+
+    /**
+     * Add a checklist to card.
```
Comment (raulchoque-eng) [inline src/main/java/trello/ui/pages/modal/CheckListModal.java]: Please, change the verb in third person.

File: src/main/java/trello/ui/pages/modal/CheckListModal.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+     *
+     * @param checkListTitle defines the title od the new checklist.
+     */
+    public void addCheckList(final String checkListTitle) {
+        WebDriverMethod.setTxtElement(checkListTitleField, checkListTitle);
+        clickConfirmAddCheckListButton();
+    }
+
+    /**
+     * Wait that add button to be clickable.
```
Comment (raulchoque-eng) [inline src/main/java/trello/ui/pages/modal/CheckListModal.java]: Please, change the verb in third person.

File: src/test/java/trello/steps/CardSteps.java [hunk 6]
```diff
@@ -71,4 +75,41 @@ public void verifyTheNameCard() {
         String actualTitle = boardPage.getCardTitle(context.getList().getTitle(), context.getCard().getTitle());
         Assert.assertEquals(context.getCard().getTitle(), actualTitle);
     }
+
+    /**
+     * Select a card of board.
```
Comment (raulchoque-eng) [inline src/test/java/trello/steps/CardSteps.java]: Please, change the verb in third person.

File: src/test/java/trello/steps/CardSteps.java [hunk 7]
```diff
@@ -71,4 +75,41 @@ public void verifyTheNameCard() {
...
+     */
+    @And("I select (.*) card")
+    public void selectCard(final String cardTitle) {
+        boardPage = new BoardPage();
+        context.getCard().setTitle(cardTitle);
+        boardPage.selectedCard(cardTitle);
+    }
+
+    /**
+     * Add a checklist to card.
```
Comment (raulchoque-eng) [inline src/test/java/trello/steps/CardSteps.java]: Please, change the verb in third person.

File: src/main/java/trello/ui/pages/BoardPage.java [hunk 8]
```diff
@@ -168,11 +175,24 @@ public String getCardTitle(final String listTitle, final String cardTitle) {
...
+    public void selectedCard(final String cardTitle) {
+        for (int i = 0; i < cards.size(); i++) {
+            if (cards.get(i).getText().equals(cardTitle)) {
+                cards.get(i).click();
+            }
+        }
+    }
+
     /**
      * Wait until Page object is found.
```
Comment (raulchoque-eng) [inline src/main/java/trello/ui/pages/BoardPage.java]: Please, change the verb in third person.

File: src/main/java/trello/ui/pages/modal/CardModal.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    /**
+     * Searches a checklist in card.
+     *
+     * @param checkListTitle defines the checklist title that search.
+     * @return a boolean with true if exists.
+     */
+    public boolean searchCheckList(final String checkListTitle) {
+        boolean result = false;
+        for (WebElement checkList : checkLists) {
+            if (checkList.findElement(By.cssSelector("h3.hide-on-edit")).getText().equals(checkListTitle)) {
```
Comment (raulchoque-eng) [inline src/main/java/trello/ui/pages/modal/CardModal.java]: 1.- Please, get out the "h3.hide-on-edit" to a variable of class.
2.- Please, use the "By.className" of account of "By.cssSelector".

File: src/main/java/trello/ui/pages/BoardPage.java (hunk 1 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/ui/pages/BoardPage.java]: Solved

File: src/main/java/trello/ui/pages/modal/CardModal.java (hunk 2 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/ui/pages/modal/CardModal.java:31]: Solved

File: src/main/java/trello/ui/pages/modal/CardModal.java (hunk 3 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/ui/pages/modal/CardModal.java]: Solved

File: src/main/java/trello/ui/pages/modal/CheckListModal.java (hunk 4 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/ui/pages/modal/CheckListModal.java]: Solved

File: src/main/java/trello/ui/pages/modal/CheckListModal.java (hunk 5 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/ui/pages/modal/CheckListModal.java]: Solved

File: src/test/java/trello/steps/CardSteps.java (hunk 7 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/trello/steps/CardSteps.java]: Solved

File: src/main/java/trello/ui/pages/BoardPage.java (hunk 8 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/ui/pages/BoardPage.java]: Solved

File: src/test/java/trello/steps/CardSteps.java (hunk 6 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/trello/steps/CardSteps.java]: Solved

File: src/main/java/trello/ui/pages/modal/CardModal.java (hunk 9 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/ui/pages/modal/CardModal.java]: Solved

File: src/main/java/trello/ui/pages/modal/CardModal.java [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+        for (WebElement checkList : checkLists) {
+            if (checkList.findElement(By.className(CARD_TITLE)).getText().equals(checkListTitle)) {
+                result = true;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Wait that card title is visibility.
```
Comment (raulchoque-eng) [inline src/main/java/trello/ui/pages/modal/CardModal.java]: Please, change this verb to third person.

Comment (JosueRodriguezGarcia) [reply src/main/java/trello/ui/pages/modal/CardModal.java]: Solved

**Review by raulchoque-eng — APPROVED** [review]

**Pull request merged**